src/arviz_plots/plots/lm_plot.py

```diff
@@ -0,0 +1,412 @@
+"""lm plot code."""
+
+import warnings
+from collections.abc import Mapping, Sequence
+from copy import copy
+from importlib import import_module
+from typing import Any, Literal
+
+import arviz_stats as azs
+import numpy as np
+import xarray as xr
+from arviz_base import rcParams
+from arviz_base.labels import BaseLabeller
+
+from arviz_plots.plot_collection import PlotCollection
+from arviz_plots.plots.utils import (
+    filter_aes,
+    get_contrast_colors,
+    get_group,
+    process_group_variables_coords,
+    set_wrap_layout,
+)
+from arviz_plots.visuals import fill_between_y, labelled_x, labelled_y, line_xy, scatter_xy
+
+
+def plot_lm(
+    dt,
+    y=None,
+    x=None,
+    y_pred=None,
+    x_pred=None,
+    filter_vars=None,
+    group="posterior_predictive",
+    coords=None,
+    sample_dims=None,
+    ci_kind=None,
+    ci_prob=None,
+    line_kind=None,
+    plot_collection=None,
+    backend=None,
+    labeller=None,
+    aes_by_visuals: Mapping[
+        Literal[
+            "ci_line",
+            "line",
+            "ci_fill",
+            "scatter",
+            "xlabel",
+            "ylabel",
+        ],
+        Sequence[str],
+    ] = None,
+    visuals: Mapping[
+        Literal[
+            "ci_line",
+            "line",
+            "ci_fill",
+            "scatter",
+            "xlabel",
+            "ylabel",
+        ],
+        Mapping[str, Any] | Literal[False],
+    ] = None,
+    stats: Mapping[
+        Literal["credible_interval",],
+        Mapping[str, Any] | xr.Dataset,
+    ] = None,
+    **pc_kwargs,
+):
+    """Posterior predictive and mean plots for regression-like data..
+
+    Parameters
+    ----------
+    dt : DataTree
+        Input data
+    y : str or DataArray, optional
+        Target variable. If None (default), the first variable in "observed_data" is used.
+    x : str or list of str or DataArray or Dataset, optional
+        Independent variable(s). If None (default), all variables in "constant_data" are used.
+    y_pred : str or DataArray, optional
+        Predicted values.
+        If None (default), the variable in the specified group with the same name as y is used.
+    x_pred : str or list of str or DataArray or Dataset, optional
+        Independent variable(s) for predictions.
+
+        If None (default), and if group is "predictions", all variables corresponding to x data
+        in "predictions_constant_data" group are used. If group is "posterior_predictive",
+        x is used.
+    filter_vars: {None, “like”, “regex”}, default None
+        If None (default), interpret var_names as the real variables names.
+        If “like”, interpret var_names as substrings of the real variables names.
+        If “regex”, interpret var_names as regular expressions on the real variables names.
+    group : str, default "posterior_predictive"
+        Group to use for plotting. Defaults to "posterior_predictive".
```

Comment:
```suggestion
        Group to use for plotting.
```

already indicated above